### ISSUE TITLE
[BUG]: Segmentation fault in pipeline-other with custom ONNX model via inf-triton

### ISSUE THREAD

**Issue by ysaini01** [nv-morpheus/Morpheus#2298]
### Version

25.02-runtime

### Which installation method(s) does this occur on?

Docker

### Describe the bug.

I am trying to run inference in Morpheus pipeline-other using my own ONNX model deployed on Triton Inference Server. The pipeline builds successfully but crashes with a segmentation fault in TensorMemory::has_tensor() when running the inf-triton stage. The same model works when tested directly via curl against Triton. The issue seems specific to Morpheus integration. 

My triton logs looks like this ( _From the Triton container, I can see only health and metadata requests coming through. It never reaches the actual inference stage (/v2/models/my_model/infer)_ ):
```
I0826 12:08:39.441420 1 http_server.cc:4590] "HTTP request: 0 /v2/health/live"
I0826 12:08:39.581532 1 http_server.cc:4590] "HTTP request: 0 /v2/health/live"
I0826 12:08:39.582230 1 http_server.cc:4590] "HTTP request: 0 /v2/health/live"
I0826 12:08:39.582955 1 http_server.cc:4590] "HTTP request: 0 /v2/health/live"
I0826 12:08:39.583463 1 http_server.cc:4590] "HTTP request: 0 /v2/models/my_model/ready"
I0826 12:08:39.583496 1 model_lifecycle.cc:339] "GetModel() 'my_model' version -1"
I0826 12:08:39.583990 1 http_server.cc:4590] "HTTP request: 0 /v2/models/my_model"
I0826 12:08:39.584016 1 model_lifecycle.cc:339] "GetModel() 'my_model' version -1"
I0826 12:08:39.584035 1 model_lifecycle.cc:297] "VersionStates() 'my_model'"
I0826 12:08:39.584523 1 http_server.cc:4590] "HTTP request: 0 /v2/models/my_model/config"
I0826 12:08:39.584569 1 model_lifecycle.cc:339] "GetModel() 'my_model' version -1"
```

### Minimum reproducible example

```shell
Here is the exact command I am running inside the Morpheus container:

morpheus --log_level=DEBUG run pipeline-other \
  from-file --filename=/workspace/morpheus_pipeline/input/input.csv \
  deserialize \
  inf-triton \
    --model_name=my_model \
    --server_url=10.221.80.1:8200 \
    --force_convert_inputs=True \
  serialize \
  to-file --filename=/workspace/morpheus_pipeline/output/output.csv --overwrite
```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

Configuring Pipeline via CLI
Module 'FileBatcher' was successfully registered with 'morpheus' namespace.
Module 'FileToDF' was successfully registered with 'morpheus' namespace.
Module 'FilterCmFailed' was successfully registered with 'morpheus' namespace.
Module 'FilterControlMessage' was successfully registered with 'morpheus' namespace.
Module 'FilterDetections' was successfully registered with 'morpheus' namespace.
Module 'FromControlMessage' was successfully registered with 'morpheus' namespace.
Module 'MLFlowModelWriter' was successfully registered with 'morpheus' namespace.
Module 'PayloadBatcher' was successfully registered with 'morpheus' namespace.
Module 'Serialize' was successfully registered with 'morpheus' namespace.
Module 'ToControlMessage' was successfully registered with 'morpheus' namespace.
Module 'WriteToElasticsearch' was successfully registered with 'morpheus' namespace.
Module 'WriteToFile' was successfully registered with 'morpheus' namespace.
Module 'deserialize' was successfully registered with 'morpheus' namespace.
W20250826 11:41:53.889091 133091194566208 thread.cpp:137] unable to set memory policy - if using docker use: --cap-add=sys_nice to allow membind
====Pipeline Pre-build====
====Pre-Building Segment: linear_segment_0====
====Pre-Building Segment Complete!====
====Pipeline Pre-build Complete!====
====Registering Pipeline====
Starting pipeline via CLI... Ctrl+C to Quit
====Building Pipeline====
====Building Pipeline Complete!====
====Registering Pipeline Complete!====
Config: 
{
  "_model_max_batch_size": 8,
  "_pipeline_batch_size": 256,
  "ae": null,
  "class_labels": [],
  "debug": false,
  "edge_buffer_size": 128,
  "execution_mode": "GPU",
  "feature_length": 1,
  "fil": {
    "feature_columns": null
  },
  "frozen": true,
  "log_config_file": null,
  "log_level": 10,
  "mode": "OTHER",
  "num_threads": 64,
  "plugins": []
}
CPP Enabled: True
====Starting Pipeline====
====Pipeline Started====
====Building Segment: linear_segment_0====
Added source: <from-file-0; FileSourceStage(filename=/workspace/morpheus_pipeline/input/input.csv, iterative=False, file_type=FileTypes.Auto, repeat=1, filter_null=True, filter_null_columns=(), parser_kwargs={})>
  └─> morpheus.MessageMeta
Added stage: <deserialize-1; DeserializeStage(ensure_sliceable_index=True, task_type=None, task_payload=None)>
  └─ morpheus.MessageMeta -> morpheus.ControlMessage
Added stage: <inference-2; TritonInferenceStage(model_name=my_model, server_url=10.221.80.1:8200, force_convert_inputs=True, use_shared_memory=False, needs_logits=None, inout_mapping={}, input_mapping={}, output_mapping={}, thread_count=None)>
  └─ morpheus.ControlMessage -> morpheus.ControlMessage
Added stage: <serialize-3; SerializeStage(include=(), exclude=(), fixed_columns=True)>
  └─ morpheus.ControlMessage -> morpheus.MessageMeta
Added stage: <to-file-4; WriteToFileStage(filename=/workspace/morpheus_pipeline/output/output.csv, overwrite=True, file_type=FileTypes.Auto, include_index_col=True, flush=False)>
  └─ morpheus.MessageMeta -> morpheus.MessageMeta
====Building Segment Complete!====
*** Aborted at 1756208514 (unix time) try "date -d @1756208514" if you are using GNU date ***
PC: @     0x790c26571ad2 morpheus::TensorMemory::has_tensor(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const
*** SIGSEGV (@0x20) received by PID 15581 (TID 0x790a637f2640) from PID 32; stack trace: ***
    @     0x790e5c114ee8 (/usr/lib/x86_64-linux-gnu/libc.so.6+0x99ee7)
    @     0x790c34ca3917 google::(anonymous namespace)::FailureSignalHandler(int, siginfo*, void*)
    @     0x790e5c0bd520 (/usr/lib/x86_64-linux-gnu/libc.so.6+0x4251f)
    @     0x790c26571ad2 morpheus::TensorMemory::has_tensor(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const
    @     0x790c266498f5 morpheus::InferenceClientStage::on_data(morpheus::InferenceClientStage::on_data(std::shared_ptr<morpheus::ControlMessage>&&, std::shared_ptr<mrc::coroutines::Scheduler>)::_ZN8morpheus20InferenceClientStage7on_dataEOSt10shared_ptrINS_14ControlMessageEES1_IN??*
    @     0x790c26646aeb pybind11::cpp_function::initialize<mrc::pymrc::AsyncioScheduler::resume(mrc::pymrc::PyObjectHolder, std::__n4861::coroutine_handle<void>)::{lambda()#1}, void>(mrc::pymrc::AsyncioScheduler::resume(mrc::pymrc::PyObjectHolder, std::__n4861::coroutine_handle<v??*
    @     0x790c2650802c pybind11::cpp_function::dispatcher(_object*, _object*, _object*)
    @     0x618762639c08 cfunction_call
    @     0x61876261d17c _PyObject_MakeTpCall.localalias
    @     0x6187625de105 context_run
    @     0x61876262ff0b cfunction_vectorcall_FASTCALL_KEYWORDS
    @     0x6187625316f0 _PyEval_EvalFrameDefault.cold
    @     0x618762662d26 method_vectorcall
    @     0x790c26541be5 pybind11::detail::simple_collector<(pybind11::return_value_policy)1>::call(_object*) const
    @     0x790c2664cec3 mrc::pymrc::AsyncioRunnable<std::shared_ptr<morpheus::ControlMessage>, std::shared_ptr<morpheus::ControlMessage> >::run(mrc::runnable::Context&)
    @     0x790c264eda8f mrc::runnable::RunnableWithContext<mrc::runnable::Context>::main(mrc::runnable::Context&)
    @     0x790c29bc4a5e std::_Function_handler<void (), mrc::runnable::Runner::enqueue(std::shared_ptr<mrc::runnable::IEngines>, std::vector<std::shared_ptr<mrc::runnable::Context>, std::allocator<std::shared_ptr<mrc::runnable::Context> > >&&)::{lambda()#1}>::_M_invoke(std::_Any_X?n
    @     0x790c29ad7488 std::thread::_State_impl<std::thread::_Invoker<std::tuple<mrc::system::ThreadResources::make_thread<boost::fibers::packaged_task<void ()> >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, mrc::CpuSet, boost::fibers::packageX?n
    @     0x790c35a80b6d execute_native_thread_routine
    @     0x790e5c10fac3 (/usr/lib/x86_64-linux-gnu/libc.so.6+0x94ac2)
    @     0x790e5c1a0a04 clone
Segmentation fault (core dumped)

</pre></details>

### Full env printout

<details><summary>Click here to see environment details</summary><pre>

- Morpheus version: 25.02-runtime
- Python version: 3.12.3
- CUDA version: 12.9
- Triton Inference Server: 25.02
- ONNX Runtime: N/A (using Triton only)
- OS: Ubuntu 24.04 LTS
- Running inside Docker: Yes (morpheus container)

</pre></details>

### Other/Misc.

Things I have tried:

- Verified Triton container is running (morpheus-tritonserver-models:25.02).
- Confirmed Triton is reachable via curl to v2/health/ready and model inference works outside Morpheus.
- Tested with both pipeline-nlp and pipeline-other commands.
- Tried different model configurations and formats (onnx model with single input/output).
- Ran docker logs on Triton container to ensure model repository is correctly loaded.
- Tried adjusting --input_mapping and --output_mapping in inf-triton stage.

### Code of Conduct

- [x] I agree to follow Morpheus' Code of Conduct
- [x] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

**Comment by dagardner-nv** [comment]
@ysaini01 a failure on the `has_tensor` method implies that the input message is missing an expected tensor field required by the `input_mapping` specified.

Based on the command you provided the pipeline layout is :
```
from-file --> deserialize --> inf-triton --> serialize --> to-file
```

Typically in a Morpheus pipeline a pre-process stage is used after the `deserialize` stage and before the `inf-triton` which copies (and potentially transforms) data from input fields in the dataframe into a tensor. 

Take a look at the examples https://github.com/nv-morpheus/Morpheus/tree/branch-25.02/examples
find the one that most closely matches what you are attempting to accomplish and take a look at the pre-process stage in used. A python examples can be found at :
https://github.com/nv-morpheus/Morpheus/blob/branch-25.02/examples/abp_pcap_detection/abp_pcap_preprocessing.py
